zerver/lib/markdown/__init__.py

```diff
@@ -1989,7 +1990,7 @@ def handleMatch(  # type: ignore[override] # https://github.com/python/mypy/issu
 
                 if not silent:
                     self.zmd.zulip_rendering_result.mentions_user_group_ids.add(user_group.id)
-                name = user_group.name
+                name = get_user_group_mention_display_name(user_group)
```

Comment:
Hmm, this is a bit annoying, because these groups names are tagged for translation; will this be the right group name for all users?

I think it might be that we need this translation layer to happen in all clients, not in the server, and we need an API design conversation about how to do that nicely. It should be OK to have `role:members` in the raw markup, and the `rendered_markdown.ts` logic rewrite it.

Reply:
Oh, I see, you are doing that in `rendered_markdown.ts`, so this name is just for clients that don't support that. OK.

(Maybe we discussed thus, and I've forgotten the details).

Reply:
I don't get the last question clearly. `render_markdown.ts` logic uses the ID to first get the group object and then the group name, like it does to get the latest name of the group.

And regarding right group name for all users for the names returned by server, we return translated error messages as well from the server, so that should work I guess.

Reply:
Ahh, right, I was somehow thinking about raw syntax, not rendered HTML. So I think this is OK.